internal/controller/clusterinstance/template_engine.go

```diff
@@ -55,8 +56,16 @@ func (te *TemplateEngine) ProcessTemplates(
 	var renderedObjects RenderedObjectCollection
 	log.Info("Started processing cluster-level install templates")
 
+	// Copy ClusterImageSet from ClusterInstance
+	clusterImageSet := hivev1.ClusterImageSet{}
+	key := types.NamespacedName{Name: clusterInstance.Spec.ClusterImageSetNameRef, Namespace: ""}
+	if err := c.Get(ctx, key, &clusterImageSet); err != nil {
+		log.Error("Encountered error while processing cluster image set reference", zap.Error(err))
+		return renderedObjects, err
```

Comment:
This will have a nice side-effect of catching a typo in the ClusterImageSet sooner, too!

Reply:
There is a check in the validating webhook already, but I think the linter and code coverage bots would have my head if I ignored the error value here.